src/components/Routes/Find/Search/Search.js

```diff
@@ -103,7 +103,8 @@ class Search extends Component {
     event.preventDefault();
 
     this.props.searchPaginationChangeDone(0, 1);
-    this.props.searchGo(userIp, userAgent, 0, limit, sortBy, this.state.form.query.value, this.state.form.location.value, this.state.form.country.value, this.state.form.radius.value, this.state.form.jobType.value, this.state.form.age.value);
+    this.props.searchSortByChangeDone('relevance');
+    this.props.searchGo(userIp, userAgent, 0, limit, 'relevance', this.state.form.query.value, this.state.form.location.value, this.state.form.country.value, this.state.form.radius.value, this.state.form.jobType.value, this.state.form.age.value);
```

Comment:
If `searchGo` was refactored to accept an object argument, this could be simplified to:

```
const formValues = Object.keys(this.state.form).reduce(
  (formValuesByKey, formKey) => ({
    ...formValuesByKey,
    [formKey]: this.state.form[formKey].value,
  }),
  {}
);

this.props.searchGo({
  userIp,
  userAgent,
  limit: 0,
  sortBy: 'relevance',
  ...formValues,
});
```

You could then extract that reduce to a function and move it to utilities:

```
const getFormValuesFromState = (state) => Object.keys(this.state.form).reduce(
  (formValuesByKey, formKey) => ({
    ...formValuesByKey,
    [formKey]: this.state.form[formKey].value,
  }),
  {}
);
```

which would be invoked like this:
```
const formValues = getFormValuesFromState(this.state.form);
```


Moving the complexity of the internals away from where you're actually dispatching the action means a few things:

- You don't need to think about how `getFormValuesFromState` works, unless you're changing it
- The call of `searchGo` reads exactly as you want it to execute
- There is less "visual noise" or "surface area" to scan
- You're freed up to reason about what's going on at this level of abstraction, rather than having to understand everything at each level all at once
  - This is one of the reasons that the Flux/Redux pattern is so nice to work with. You keep your business logic isolated away in a different place, allowing your component logic to just *use* it

This concept is referred to by a few names, but generally can be thought of as "boundaries".

Say you wanted to customize the list of form values that `searchGo` receives. It now lives in a separate boundary, away from the actual dispatcher invocation, and away from `searchGo`'s definition.

```
const formKeysWhitelist = [
  'query',
  'location',
  'country',
  'radius',
  'jobType',
  'age',
];

const getFormValuesFromState = (state) => Object.keys(this.state.form)
  .filter((formKey) => formKeysWhitelist.includes(formKey))
  .reduce(
    (formValuesByKey, formKey) => ({
      ...formValuesByKey,
      [formKey]: this.state.form[formKey].value,
    }),
    {}
  );
```

So, maybe `formKeysWhitelist` lives co-located with `getFormValuesFromState`, maybe it lives somewhere else in a global config, maybe it sits in the utils file next to `getFormValuesFromState`, but gets exported. It all comes down to matching levels of abstraction with one another; finding the right "boundary" where something should live and keeping it there. Any questions?